### PR TITLE
[#3598] constrain resource last_changed_by field to be not null and default t…

### PR DESCRIPTION
**Pull request by sblack-usu**
#3598

Constrains the last_changed_by field on the resource to non null and defaults the last_changed_by to the admin user.  

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Try to view a resource with last_changed_by set to None.  Verify you can't, run the migration in this command, then open the resource again.  Verify the resource opens and the last_changed_by user is admin.

### CODE REVIEW

**Comment by sblack-usu** [comment]
@hyi and @alvacouch - I'd like both of you to take a look at this.  It may be controversial to default the last_changed_by user to admin.  I'm open to other approaches if you have any.

I think it's also worth pointing out that this change will cause an error in the workflow that caused last_changed_by to be set to None.  I think it was probably a management command (I can't be sure of this) so users probably won't experience an error.

**Comment by hyi** [comment]
@sblack-usu Setting None last_changed_by to admin sounds reasonable to me. Another option would be to set to the first creator of the resource but it most likely does not reflect the fact. I'd vote for setting it to admin since it at least reflects the fact, and it the resource gets modified again, a valid user would override the admin user, which sounds reasonable to me.

File: hs_core/hydroshare/utils.py [hunk 1]
```diff
@@ -469,7 +469,10 @@ def resource_modified(resource, by_user=None, overwrite_bag=True):
 
     """
 
-    resource.last_changed_by = by_user
+    if not by_user:
+        logger.warning("by_user not specified in resource_modified, last_chagned_by will not be updated")
```
Comment (alvacouch) [inline hs_core/hydroshare/utils.py]: Spelling chagned --> changed

**Comment by sblack-usu** [comment]
retest this, please

**Comment by sblack-usu** [comment]
retest this, please

**Comment by sblack-usu** [comment]
retest this, please

**Comment by sblack-usu** [comment]
retest this, please

**Comment by sblack-usu** [comment]
reopening in a squashed branch in PR #3846

**Pull request closed without merging**